### PR TITLE
Add Client::wait_for_transaction_acceptance

### PR DESCRIPTION
**Pull request by thibault-martinez**

### CODE REVIEW

File: sdk/src/client/api/wait_for_tx_acceptance.rs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    client::{Client, Error},
+    types::{api::core::TransactionState, block::payload::signed_transaction::TransactionId},
+};
+
+pub(crate) const DEFAULT_WAIT_FOR_TX_ACCEPTANCE_INTERVAL: Duration = Duration::from_millis(500);
+pub(crate) const DEFAULT_WAIT_FOR_TX_ACCEPTANCE_MAX_ATTEMPTS: u64 = 80;
+
+impl Client {
+    /// Checks the transaction state for a provided transaction id until it's accepted. Interval in milliseconds.
+    /// Returns the block id that contains this transaction.
```
Comment (Thoralf-M) [inline sdk/src/client/api/wait_for_tx_acceptance.rs]:
```suggestion
```

Comment (Thoralf-M) [reply sdk/src/client/api/wait_for_tx_acceptance.rs]: Also needs to be removed from the wallet fn

File: bindings/python/iota_sdk/wallet/wallet.py [hunk 2]
```diff
@@ -720,17 +720,17 @@ def prepare_transaction(
         return PreparedTransaction(self, prepared)
 
     def wait_for_transaction_acceptance(
-            self, transaction_id: TransactionId, interval=None, max_attempts=None) -> BlockId:
+            self, transaction_id: TransactionId, interval=None, max_attempts=None):
         """Checks the transaction state for a provided transaction id until it's accepted. Interval in milliseconds. Returns the block id that
         contains this transaction.
```
Comment (Thoralf-M) [inline bindings/python/iota_sdk/wallet/wallet.py]:
```suggestion
        """Checks the transaction state for a provided transaction id until it's accepted. Interval in milliseconds.
```

File: sdk/src/client/api/wait_for_tx_acceptance.rs (hunk 1 above)
Comment (thibault-martinez) [reply sdk/src/client/api/wait_for_tx_acceptance.rs]: Done

File: bindings/python/iota_sdk/wallet/wallet.py (hunk 2 above)
Comment (thibault-martinez) [reply bindings/python/iota_sdk/wallet/wallet.py]: Done

**Review by DaughterOfMars — DISMISSED** [review]

File: bindings/python/examples/how_tos/nft_collection/01_mint_collection_nft.py [hunk 3]
```diff
@@ -65,10 +65,11 @@ def get_immutable_metadata(index: int) -> str:
     transaction = wallet.mint_nfts(chunk)
 
     # Wait for transaction to get accepted
```
Comment (Thoralf-M) [inline bindings/python/examples/how_tos/nft_collection/01_mint_collection_nft.py]: Any point in keeping all these comments?

Comment (thibault-martinez) [reply bindings/python/examples/how_tos/nft_collection/01_mint_collection_nft.py]: Probably not

Comment (thibault-martinez) [reply bindings/python/examples/how_tos/nft_collection/01_mint_collection_nft.py]: Removed

**Review by Thoralf-M — APPROVED** [review]

**Review by DaughterOfMars — APPROVED** [review]

**Pull request merged**